### ISSUE TITLE
sit on stairs without command??

### ISSUE THREAD

**Issue by ScuroK** [Zrips/CMI#256]
How can i allow players to sit on stairs by a single click?? i gave them cmi.command.sit and cmi.command.sit.stairs but it wont work. but i can do it. so what is the right permission??

**Comment by ScuroK** [comment]
also douple-klick does not help. non-op players cant sit without /sit

**Comment by MCTyler** [comment]
Hmm, haven't tried lately and Server not up but dbl-click, very fast used to work....
###
Animations:
  # Enable siting on stair block by clicking on them with empty hand or by looking and using command
  # Requires cmi.command.sit.stairs
  SitOnStairs: true
  StairsAsChairs: true
  SlabsAsChairs: true
  # Player will sit on chair only after rapid double click
  DoubleClick: true
  # Delay in milliseconds between clicks to sit on chair when double click is enabled
  DoubleClickDelay: 200
###

**Comment by ScuroK** [comment]
did it all. no success!

**Comment by MCTyler** [comment]
Let me fire up the server...Might be a problem with latest update.

**Comment by ScuroK** [comment]
i think so. i was only able to give them /sit permissions.
but i dont like commands fpr this. a click would be good enough.

further i found if double-click is disabled in the config, i only need a single click as OP to sit on stairs. 
i guess there is a mess or a missing permission for click-sitting.

will check back tmr. :)

**Comment by ScuroK** [comment]
Animations:
  # Enable siting on stair block by clicking on them with empty hand or by looking and using command
  # Requires cmi.command.sit.stairs
  SitOnStairs: true
  StairsAsChairs: true
  SlabsAsChairs: true
  # Player will sit on chair only after rapid double click
  DoubleClick: true
  # Delay in milliseconds between clicks to sit on chair when double click is enabled
  DoubleClickDelay: 200

I gave cmi.command.sit.stairs to the default group and i even restarted the server. Still players cant sit using double click.

**Comment by ScuroK** [comment]
@Zrips any ideas?

**Comment by ScuroK** [comment]
im not sure if this is related but i found this in my log:
[11:03:27] [Server thread/INFO]: jaenu1 issued server command: /spawn
[11:03:28] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m3400[m
[11:03:29] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m3500[m
[11:03:30] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m3600[m
[11:03:31] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m3700[m
[11:03:32] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m3800[m
[11:03:33] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m3900[m
[11:03:34] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m4000[m
[11:03:35] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m4100[m
[11:03:36] [Server thread/INFO]: [0;33;22mChecking... [0;33;1m4752[0;33;22m/[0;33;1m4200[m

**Comment by ScuroK** [comment]
and i also found:
[11:02:26] [Server thread/WARN]: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[11:02:26] [Server thread/WARN]: SLF4J: Defaulting to no-operation (NOP) logger implementation
[11:02:26] [Server thread/WARN]: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

**Comment by Zrips** [comment]
@ScuroK you need permission and you need empty hands. Check what blocks are valid for chairs and that player has access to required permission node by using /cmi haspermission [playerName] [permissionNode]

**Comment by ScuroK** [comment]
@zrips i did give them cmi.command.sit.stairs and it didnt work. It only works for OPs.

Any hidden permission i dont know??

**Comment by Zrips** [comment]
Looking throw code, it could be that because player cant teleport in that location, he cant sit there

**Comment by ScuroK** [comment]
how do you mean? i did not block teleportation in this area (using worldguard or anything). And i also tryed it in different worlds, gamemodes and areas.

**Comment by MCTyler** [comment]
I have noticed too that only a empty hand can single-click (right) and sit on stairs.

**Comment by ScuroK** [comment]
Guys, i tried it all! And im sure i tested it the right way because it works when i use an oppen account!

**Comment by CloudeLecaw** [comment]
1. they Need the permision
2. NO items in Hands
3. klick on stairs

**Comment by Zrips** [comment]
@ScuroK what feedback are you getting with **/cmi haspermission [playerName] cmi.command.sit.stairs**

**Comment by LogGits** [comment]
@ScuroK works for me.
Correct me if im wrong. 

You are using luckperms. The user is in a group that has access to `cmi.command.sit.stairs`?
The user is rightclicking the stair without holding shift?
These values are set correctly:
```
  SitOnStairs: true
  StairsAsChairs: true
```

If you have done all this, does it give you any feedback when they try to rightclick with an empty slot? Do they have items in their other hand? Have you tried testing it yourself as a user in that group? What version of spigot is your server running and specify if it's paperspigot? 
(need this info to test)

**Comment by ScuroK** [comment]
I use latest spigot and latest.

No feedback.

Yes i did test it myself

Yes i did all as you described.

Maybe its a plugin conflict?
I use DynamicStands (vehicles)
And MyPet (riding)

Guys pls dont tell me again how it works. I realy did it this way...

**Comment by ScuroK** [comment]
And no its not paper spigot

**Comment by ScuroK** [comment]
@mercurialmusic do you use mypet or Dynamic Stands or any other riding relatef plugins?

**Comment by ScuroK** [comment]
@Zrips it tells me „hard set to True by some plugin“

**Comment by mercurialmusic** [comment]
We do use MyPet, but we've disabled the riding for all pets. We also have [PiggyBack](https://www.spigotmc.org/resources/piggyback-more-than-just-a-stacker.14130/), which is the opposite of riding (the mobs ride the player), and then of course the /cmi ride. Haven't had issues with those, though -- I never expected that they would interfere.

**Comment by ScuroK** [comment]
i also think they should not interfere.

**Comment by ScuroK** [comment]
i hope he will fix it!
btw i dont think its because of mypet. otherwise itwould not work as OP.

**Comment by mercurialmusic** [comment]
One thing I've noticed is that ScuroK and I both use LuckPerms, which I know CMI hasn't always got along with and might be the offender here, in which case a fix shouldn't affect anyone else. Out of curiosity, @MCTyler what permissions plugin do you use?

**Comment by ScuroK** [comment]
Hmm yeah but LP is the new standard.

**Comment by MCTyler** [comment]
LuckPerms...Who would use anything else? :)

**Comment by MCTyler** [comment]
CMI handles everything Essentials used to Plus Economy. I have no other Essentials junk installed.

**Comment by ScuroK** [comment]
What spigot build are you using? Is it up to date?

**Comment by MCTyler** [comment]
> ver
[14:27:01 INFO]: This server is running Paper version git-Paper-1390 (MC: 1.12.2) (Implementing API version 1.12.2-R0.1-SNAPSHOT)
[14:27:01 INFO]: Checking version, please wait...
[14:27:01 INFO]: You are 1 version(s) behind

**Comment by ScuroK** [comment]
the bug MUST be some permission problem. otherwise it would not work for OPs

**Comment by MCTyler** [comment]
Did you enable ```/lp verbose on ```  and see what it's throwing?

**Comment by ScuroK** [comment]
can you please show me the lp-permissions-file of your default group?

**Comment by MCTyler** [comment]
https://luckperms.github.io/editor/?zoegIdkOaq

**Comment by ScuroK** [comment]
hmmm i dont see anything special. all i know is that it doesnt work for me :(

@Zrips any news?

**Comment by CloudeLecaw** [comment]
hm on my Server i would use my permissionscanner for check all used/asked permissions when rightclicking the stair
this whyu also seeif a perm is negated andthen u can test it
![image](https://user-images.githubusercontent.com/12080017/38721006-c6fe5ea6-3ef8-11e8-8d70-08f5cad666d8.png)

u see to permsisions are asked for
**cmi.teleport.with.armor_stand**   and **cmi.command.sit.stairs**

pls test it

**Comment by CloudeLecaw** [comment]
i agree CMI got this big, but i also see this massive amount oftime it would cost to create a full wiki for it :3

**Comment by ScuroK** [comment]
Thank you very much! I am currently at work and very busy. But i will write back later 😊

**Comment by ScuroK** [comment]
It might be because of the plugin DynamicStands. It offers control over armorstands and it includes armorstand-based vehicles (=riding armorstands)
Later will test it without DynamicStands and see what happens.

**Comment by MCTyler** [comment]
A lot of these plugins use armor stands to accomplish what they need in a Sudo Fasion.

**Comment by ScuroK** [comment]
But i dont think so because it works for ops

**Comment by CloudeLecaw** [comment]
yes but the other Plugins also have permissions that can be given or denied ;)

**Comment by CloudeLecaw** [comment]
i tried - cmi.command.sit.stairs on a Spigot + myPermissionsystem + CMI
- cmi.command.sit.stairs
and gave no other permissions.


then i placed a stair in the world and rightcliked it with empty Hand -> my char is sitting

![image](https://user-images.githubusercontent.com/12080017/38726884-bd3489dc-3f0b-11e8-8870-7b768c960b96.png)

**Comment by ScuroK** [comment]
DynamicStands has no impact.
continue testing

**Comment by ScuroK** [comment]
Removed MyPet: No impact
Removed DynamicStands: No impact

**Comment by MCTyler** [comment]
Might consider installing plugman for this test and remove afterwords.

**Comment by CloudeLecaw** [comment]
@ MCTyler i settet up a sever with all Plugins u listet
only one i did not got are
- AutomaticInventory
- InstaWall,
- LuckPerms (i use my own u know^^)
- mcMMO,
- (mobhunting -may outdated Version...)

![image](https://user-images.githubusercontent.com/12080017/38728640-ba7b138c-3f10-11e8-9a85-2621bc6265ba.png)
but finnaly non of These Plugin waseffecting the sitting :3

![image](https://user-images.githubusercontent.com/12080017/38728697-db92817c-3f10-11e8-8ec9-3a90ea7ad24d.png)

**Comment by MCTyler** [comment]
Yes I know, Mine works fine, it's ScuroK were trying to help.

**Comment by ScuroK** [comment]
![2018-04-13_14 00 54](https://user-images.githubusercontent.com/10480144/38733699-2e6a9828-3f23-11e8-82c0-9c02ec150b8a.png)

**Comment by CloudeLecaw** [comment]
ui These are a lot Plugins.

may u want to invite me to your discord? or your simply post your Server adress? :-)

**Comment by MCTyler** [comment]
Skript? Yuk! that messes up a lot of things.

**Comment by CloudeLecaw** [comment]
Yes Tyler but ist his Server so he can use as much Plugins at he wants ;)

**Comment by MCTyler** [comment]
Just saying...had issues with that interfering with stuff in the past...were trying to fix sit not judge.

**Comment by ScuroK** [comment]
my plugins are well balanced and i created this set of plugins careful over many years. 
i dont think blaming them is the right way to deal with it. 

if cmi is unable to work with my plugins i would expect cmi to adjust the sit function.

as i see on spigot, im not the only person having this problem:

> Am I missing something?

I have added cmi.command.sit and cmi.command.sit.stairs
and only reference in config to sitting are three things that are set to true. 

But as op i can sit, as player i cannot.. i right click stairs and they dont let players sit. 
(yes, i restarted the server)

**Comment by MCTyler** [comment]
If thats the case me and Cloud have no business commenting and its totally up to Zrips to tailor CMI to fit your server...good luck!

**Comment by ScuroK** [comment]
as for now i dont get much feedback from him. but since its not only me and my server having issues, i hope this will get fixed.

**Comment by CloudeLecaw** [comment]
Like i said, your Server your Plugins and if you say all is fine on it then all is fine :)
Your right may 1-2 ppl have this sit Problem and the reason mostly is the same
a Plugin that is Plugin it or taking the permission to fire the Event correct.

Noone said u sould use lesser Plugins ;) Some Servers use much more.

**Comment by ScuroK** [comment]
Thx. I will try to find it first

**Comment by LogGits** [comment]
Wait so when you test with just CMI and your permissions plugin and vault can you sit on stairs with rightclick?

**Comment by MCTyler** [comment]
AFAIK He has a plugin that interferes and wants Zrips to make CMI and this option work with his plugins that he has. 
```quote my plugins are well balanced and i created this set of plugins careful over many years.
i dont think blaming them is the right way to deal with it.

if cmi is unable to work with my plugins i would expect cmi to adjust the sit function.

as i see on spigot, im not the only person having this problem:

Am I missing something?

I have added cmi.command.sit and cmi.command.sit.stairs
and only reference in config to sitting are three things that are set to true.

But as op i can sit, as player i cannot.. i right click stairs and they dont let players sit.
(yes, i restarted the server)

skript does not have any scripts loaded related to armorstands or any animations.
```

**Comment by ScuroK** [comment]
Its just a wild guess

**Comment by ScuroK** [comment]
Sorry i closed by mistake. More tests will follow

**Comment by CloudeLecaw** [comment]
:) we wait for results, its still interesting to see what caused it at the end :-)

**Comment by CypherPhyre** [comment]
What version of LuckPerms are you running?

**Comment by CloudeLecaw** [comment]
CypherePhyre may i am wrong but shouldnt an owner allways try to be uptodate? im pretty Surer ScuroK is using the latest version of LuckPerms ;)

**Comment by CypherPhyre** [comment]
In an ideal world yes an Owner should, but not all servers are maintained by Owners. I am not the owner of my server. When it was first set up with LuckPerms the Owner used an outdated version. Outdated by 3 versions.

**Comment by Zrips** [comment]
@MCTyler About which exactly plugin you have been talking? Script?

**Comment by mercurialmusic** [comment]
To all in this thread, it might be good to remember that just because your experience is different doesn't invalidate someone else's. 

Looking at ScuroK's plugin list, we have very few plugins in common -- just the more popular ones that most on here would probably have -- so since I have the same issue I would not guess that it's another plugin interfering. If it is another plugin it would likely be a different one than what is interfering in ScuroK's. And someone else just recently comment on Spigot saying they have the same issue. So again, just because you don't have the issue doesn't mean someone else who does have it is trying to demand special attention.

For my players, they did not have the cmi.teleport.with.armor_stand permission, so I've added it and am also somewhat curious if that affects anything, but I won't be home to check it until later tonight.

**Comment by MCTyler** [comment]
@Zrips we haven't figured that out yet...Op has to work and will try to zero in tonite.

**Comment by CypherPhyre** [comment]
On my server, cmi.teleport.with.armor_stand permission is not used at all and there are no issues with sitting on stairs. Only have cmi.command.sit and cmi.command.sit.stairs

**Comment by ScuroK** [comment]
i continue testing now

**Comment by ScuroK** [comment]
i think its NoCheatPlus! (or my custom config of it) but let me check twice...

**Comment by mercurialmusic** [comment]
Oh shit... do you use that with MarkElf's config? I do too...

**Comment by ScuroK** [comment]
@mercurialmusic 
yes i do

**Comment by ScuroK** [comment]
but i dont get any messages from NCP. any ideas which config value might cause this??

**Comment by ScuroK** [comment]
wow, even with the default nocheat-config i cant sit. 
got some reports now:

[20:04:03 INFO]: NCP: ScuroK failed Passable: moved into a block ([blocktype]) from -177.22, 70.00, 235.51 to -178.00, 69.00, 235.00 distance 1.37 . VL 10.
[20:04:16 INFO]: NCP: ScuroK failed Passable: moved into a block ([blocktype]) from -177.19, 70.00, 235.52 to -178.00, 69.00, 235.00 distance 1.39 . VL 15.

**Comment by mercurialmusic** [comment]
I haven't the foggiest, and I never get anything from it either. I'm still at work for another couple hours, but I can look tonight too.

**Comment by ScuroK** [comment]
its not the custom configs fault i think. because even with a normal ncp setup i get this problem.

@Zrips pls fix it

**Comment by mercurialmusic** [comment]
Found this string in the NCP config: 
`    passable:
      active: true
      actions: cancel vl>21 cmdc:phase1:1:11 cancel vl>38 cmdc:phase2:1:11 cancel vl>60 cmdc:phase3:1:11 cancel`

I wonder if removing the first action, cancel vl>21 would fix it, since your VLs here are 10 & 15. Maybe move up the VL levels for the others as well.

**Comment by ScuroK** [comment]
feel free to help testing. im pretty tired and had a very bad week :S

**Comment by mercurialmusic** [comment]
Yep, will do when I get home.

**Comment by ScuroK** [comment]
you mean i should just paste this?
`passable: active: true actions: cancel vl>38 cmdc:phase2:1:11 cancel vl>60 cmdc:phase3:1:11 cancel`

**Comment by ScuroK** [comment]
oobs i didnt paste the changes into the config xD. let me try again

**Comment by ScuroK** [comment]
Nope it does not work. im turning off passable checks completely now and see what happens

**Comment by ScuroK** [comment]
also i get this error:
[20:46:17 WARN]: [NoCheatPlus] Failed to create action: NoCheatPlus doesn't know command 'phase2'. Have you forgotten to define it?
[20:46:17 WARN]: [NoCheatPlus] Failed to create action: NoCheatPlus doesn't know command 'phase3'. Have you forgotten to define it?
[20:46:17 INFO]: [NoCheatPlus] Inventory checks: FastConsume is available, disabled InstantEat.

**Comment by ScuroK** [comment]
turning off passable seems not to help. something is very weird here :D

**Comment by ScuroK** [comment]
NCP also fires "survival fly" warnings... im getting pissed.

**Comment by CypherPhyre** [comment]
If you have not already done so, fully disable NCP and test to be 100% sure that it is what causes the issues. If it is the cause then open an issue ticket on NCP GitHub page for added help with this. It may not be limited to issues with CMI. Just a thought

**Comment by ScuroK** [comment]
i did that arround 5 times. its deffenitely NCP

**Comment by ScuroK** [comment]
its pretty much limited to cmi afaik

**Comment by CypherPhyre** [comment]
It wouldn't hurt to seek additional help from NCP directly

**Comment by MCTyler** [comment]
Just from my experience, most people consult NCP to provide work-arounds...not the other way around.

**Comment by ScuroK** [comment]
i dont care who takes action first. i just wanna get this fixed ;)

The reason why i think @Zrips (who still didnt comment much here) should at least try to fix it is because i have used plenty of similar plugins (sit, ride, etc) and none of them were conflicting with ncp!

for example vehicle plugins use armorstands aswell. mybe it just makes more sence to find another solution for the sitting animation.

**Comment by MCTyler** [comment]
Ok, so what as Op is bypassing NCP? Does that just bypass everything? I'll be honest, that's another plugin that always messed everything up so I don't use it either.

**Comment by ScuroK** [comment]
ncp is one of the best and most used anti cheat solutions out there.

**Comment by CypherPhyre** [comment]
I don't use NCP either. Always got false possitives on my old server. My new server uses FairPlayMinecraft

**Comment by MCTyler** [comment]
I always found Observant Mods to be the best Anti cheat but that's my opinion.

**Comment by CloudeLecaw** [comment]
its normal.. maybe i told you that CMI uses not that ugly arroow in your legs sit that others use? ^^
or better..not seeable..also it works from distance in CMI.

by the way..my opinion about NCP..its a good free one out there, but not the best, it has massive false alarms eenwith hours of configuration. but to be hnest..others have this too. but there are better onces if you want to pay a few dollars ;)...but NCP is the most popular free one

**Comment by MCTyler** [comment]
Anyway its up to Zrips or the NCP jerk Devs...Good luck with them my friend :)

**Comment by ScuroK** [comment]
im not here for these kind of discussions.

**Comment by CloudeLecaw** [comment]
good to know that we now know its NCP that is causing this problem. And this is also the reason then why i have not this problem ;) It has been 4 years done now since i last really used NCP offical on a server. Since that im using others that suits more to my desires ;)...but...hmm..not free.. *g*

**Comment by MCTyler** [comment]
I would close this an open a clean issue to get rid of all this chatter.

**Comment by mercurialmusic** [comment]
To be fair, I doubt this is going to be a very high priority thing. Zrips is one of the busiest and most active plugin devs on Spigot -- lord knows my server wouldn't be half of what it is without his work. And I can't imagine complaints about his responsiveness are going to increase the priority.

**Comment by CloudeLecaw** [comment]
now we can write a ticket a the NCP developer that his plugin is causing this problem with CMI so he may find a fix :D
In my eyes Zrips did nothing wrong the wayhe release his sit feature ;)

**Comment by ScuroK** [comment]
what are you using?

**Comment by ScuroK** [comment]
nah i wont close this issue before zrips made a statement.

**Comment by CypherPhyre** [comment]
I have FairPlayMinecraft on my server

**Comment by ScuroK** [comment]
doesnt this require a special client?

**Comment by MCTyler** [comment]
Ouch $15 https://www.spigotmc.org/resources/trademe-with-api-to-create-custom-trades-1-7-10-1-12-x.7544/

**Comment by CypherPhyre** [comment]
Player must download a free app to run in order to join

**Comment by ScuroK** [comment]
no option for me.

**Comment by CypherPhyre** [comment]
Just saying it is what I use on my server

**Comment by MCTyler** [comment]
I think you will find that Zrip only responds first thing in the morning.

**Comment by ScuroK** [comment]
@mercurialmusic let me know if you find a way how to adjust the ncp config in a good way without breakting it.

**Comment by CloudeLecaw** [comment]
at the moment im fine with aac and dcm (dcm detects to fast movments and has an obfuscator in it. its one of the private plugins that were developed for my server

**Comment by ScuroK** [comment]
Looking at the reviews on the aac page and from my expirience (beside of this little problem) ncp is still my first choice. it just needs a nice config. i biught one from markelf. and i asked him to check this issue. maybe he can do something.

**Comment by Zrips** [comment]
If this is conflict with NCP then i will fix it tomorrow (its late already at my place). Its not hard to fix issue if you know cause. Its most likely due to additional move and teleport checks which can be simply disabled or some work around made. Nothing to serious and nothing to complicated to "fix". And yes, its in " as this is not bug but compatibility issue with another plugin which blocks actions.

**Comment by Zrips** [comment]
Just side note.
If you want to keep short conversations and your issue to be fixed sooner, debug if its issue from CMI or another plugin by keeping only CMI or by removing CMI and double checking if issue still there. This would save every ones time.

**Comment by MarkElf** [comment]
Hi, let's not pick apart my work and tag me in the future instead. There are two solutions, either have cmi - from what I understand allows players to sit on some block - create an exception when a player sits using ncp's api or override the block being at on with an overwrite flag. The overwrite flag will work as a temporary solution if it's just stairs being sat on though players will be able to phase through the stairs. It's a minor trade off but worth it if sitting is the priority, not that phasing through stairs is going to be noticed compared to most people ego will try to phase through a block of cobble or wood instead.

Tl-dr: Can be solved on my ncp's, I just need to know what blocks are being sat on.

**Comment by ScuroK** [comment]
@MarkElf it allows you to sit on stairs and slabs using invisible armorstands.
 The reason why i asked u to help fixing this is because beside of that cmi offers a whole ton of features, similar lile essentials. Thats why i dont want to just get rid of it.

@Zrips said he will also try to make it compatible.

**Comment by Zrips** [comment]
@ScuroK 
Next version should work properly with sit command and NCP

**Comment by asofold** [comment]
Just noticing - in case you are using the NCP API, e.g with exemptions, feel free to contact me (snippets via spigotmc pm / here), just in case there are issues/questions with other plugins using the exemption API of NCP, or if you're concerne with potential cheats arising from workarounds.
